packages/libs/error-stack/src/ext/future.rs

```diff
@@ -36,9 +32,14 @@ macro_rules! implement_future_adaptor {
 
             #[track_caller]
             fn poll(self: Pin<&mut Self>, cx: &mut TaskContext) -> Poll<Self::Output> {
-                let projection = self.project();
-                let future = projection.future;
-                let inner = projection.inner;
+                // SAFETY: The pointee of `inner` will not move. Note, that the value inside the
+                //         `Option` will be taken, but the `Option` remains in place. Additionally,
+                //         `Self` does not implement `Drop`, nor it's `#[repr(packed)]`
```

Comment:
I think this should be 
```suggestion
                //         `Self` does not implement `Drop`, nor is it `#[repr(packed)]`
```
also for the other safety comments